algorithms/src/std_algorithms/impl/Kokkos_Constraints.hpp

```diff
@@ -238,21 +238,18 @@ KOKKOS_INLINE_FUNCTION void expect_no_overlap(
     [[maybe_unused]] IteratorType2 s_first) {
   if constexpr (is_kokkos_iterator_v<IteratorType1> &&
                 is_kokkos_iterator_v<IteratorType2>) {
-    auto const view1 = first.view();
-    auto const view2 = s_first.view();
-
-    std::size_t stride1  = view1.stride(0);
-    std::size_t stride2  = view2.stride(0);
-    ptrdiff_t first_diff = view1.data() - view2.data();
+    std::size_t stride1  = first.stride();
```

Comment:
This would not work for the specialization when `is_always_strided` is false (it does not have a `stride` method).